lib/ex_doc/formatter/html/search_data.ex

```diff
@@ -2,22 +2,51 @@ defmodule ExDoc.Formatter.HTML.SearchData do
   @moduledoc false
   alias ExDoc.Utils
 
-  def create(nodes, extras, proglang) do
+  def create(nodes, extras, config) do
     items = Enum.flat_map(nodes, &module/1) ++ Enum.flat_map(extras, &extra/1)
 
-    data = %{
-      items: items,
-      content_type: "text/markdown",
-      proglang: proglang,
-      producer: %{
-        name: "ex_doc",
-        version: to_string(Application.spec(:ex_doc)[:vsn])
+    search_data =
+      %{
+        items: items,
+        content_type: "text/markdown",
+        proglang: config.proglang,
+        producer: %{
+          name: "ex_doc",
+          version: to_string(Application.spec(:ex_doc)[:vsn])
+        }
       }
-    }
 
-    ["searchData=" | ExDoc.Utils.to_json(data)]
+    search_nodes =
+      [
+        %{name: config.package, version: config.version}
+      ]
+      |> maybe_append_related_packages(config.related_packages)
+
+    [
+      "searchData=",
+      ExDoc.Utils.to_json(search_data),
+      "\nsearchNodes=",
+      ExDoc.Utils.to_json(search_nodes)
+    ]
+  end
+
+  defp maybe_append_related_packages(search_nodes, related_packages)
+       when length(related_packages) > 0 do
+    related_packages =
+      related_packages
+      |> Enum.map(fn
+        %{name: name, version: version} ->
+          %{name: name, version: to_string(version)}
+
+        name when is_atom(name) ->
+          %{name: name, version: "*"}
+      end)
+
+    Enum.concat(search_nodes, related_packages)
   end
 
+  defp maybe_append_related_packages(search_nodes, _related_packages), do: search_nodes
```

Comment:
We should revert these changes for now, I believe the related packages will come from hex.pm itself. You can emulate it though by calling `mix docs` and then customizing the docs_config.js in doc yourself. :)

Reply:
Actually, we added our own for testing:

https://github.com/elixir-lang/ex_doc/blob/8a5217e9d1287d8b198f7b686dfdc83243f043d3/mix.exs#L127

So I think you can just change the `searchNodes` there, perhaps by adding Elixir itself or `earmark_parser` to searchNodes, so you can test it too.